[reflection] Don't get parameter default values for wrapper methods

monojenkins opened the pull request:
!! This PR is a copy of mono/mono#20602,  please do not edit or review it in this repo !!<br/>Do not automatically approve this PR:<br/><br/>* Consider how the changes affect configurations in this repo,<br/>* Check effects on files that are not mirrored,<br/>* Identify test cases that may be needed in this repo.<br/><br/>!! Merge the PR only after the original PR is merged !!<br/><hr/><br/>The lookup assumes that the method came from metadata, which is not the case
for wrappers.

Fixes mono/mono#20600

vargaz commented on src/mono/mono/metadata/reflection.c:
```diff
@@ -963,7 +963,7 @@ add_parameter_object_to_array (MonoDomain *domain, MonoMethod *method, MonoObjec
 
-	if (!(sig_param->attrs & PARAM_ATTRIBUTE_HAS_DEFAULT)) {
+	if (!(sig_param->attrs & PARAM_ATTRIBUTE_HAS_DEFAULT) || method->wrapper_type != MONO_WRAPPER_NONE) {
 		if (sig_param->attrs & PARAM_ATTRIBUTE_OPTIONAL)
```
Note that dynamic methods also have wrapper_type set.

lambdageek replied:
Added a check for `MONO_WRAPPER_TYPE_DYNAMIC_METHOD`.  Verified that we don't have a crash when printing a stack trace from inside a dynamic method

lambdageek committed:
[reflection] Don't get parameter default values for wrapper methods

The lookup assumes that the method came from metadata, which is not the case
for wrappers.

Fixes mono/mono#20600
